app/models/generic_object_definition.rb

```diff
@@ -7,6 +16,18 @@ class GenericObjectDefinition < ApplicationRecord
   has_many  :generic_objects
 
   def defined_attributes
-    @defined_attributes ||= properties[:attributes].stringify_keys
+    properties[:attributes]
+  end
+
+  def properties=(props)
+    props = props.symbolize_keys
+    if props.key?(:attributes)
+      # unify attribute name as string and attribute type as symbol
+      props[:attributes] = props[:attributes].each_with_object({}) do |(name, type), memo|
```

Comment:
Prefer `h` or `hash` or something over `memo` (memo is one of those terms that just reads funny to me)